sync/sync.go

```diff
@@ -358,6 +355,37 @@ func (s *Synchronizer) pollPending(ctx context.Context, sem chan struct{}) {
 	}
 }
 
+func (s *Synchronizer) pollLatest(ctx context.Context, sem chan struct{}) {
+	poll := func() {
+		select {
+		case sem <- struct{}{}:
+			go func() {
+				highestBlock, err := s.starknetData.BlockLatest(ctx)
+				if err != nil {
+					s.log.Warnw("Failed fetching latest block", "err", err)
+				} else {
+					s.highestBlockHeader.Store(highestBlock.Header)
+				}
+				<-sem
+			}()
+		default:
+		}
+	}
+
+	ticker := time.NewTicker(time.Minute)
```

Comment:
Technically, I guess the timeout should be configurable for testing purposes. One of us can create a good-first-issue for it once this is merged.

Reply:
> Is it possible that our local head gets ahead of our "latest seen block"?

yeah it can, I didnt want to deal with concurrency issues that comes with 2 go-routines updating the `highestBlock`. But will do that.

Reply:
We should defer `<-sem` in poll tho :+1: